src/tools/UpdatePackageDependencies.ts

```diff
@@ -54,23 +68,79 @@ export function updatePackageDependencies() {
         if (dependency === undefined) {
             throw new Error(`Unable to update item for url '${url}'. No 'runtimeDependency' found with filename '${fileName}'.`);
         }
-
-        console.log(`Updating ${url}`);
         return dependency;
     };
 
-    newPrimaryUrls.split(',').forEach(urlToUpdate =>{
-        console.log(`Trying to update ${urlToUpdate}`);
-        let dependency = findDependencyToUpdate(urlToUpdate);
-        dependency.url = urlToUpdate;
+    const dottedVersionRegExp = /[0-9]+\.[0-9]+\.[0-9]+/g;
+    const dashedVersionRegExp = /[0-9]+-[0-9]+-[0-9]+/g;
+    
+    const getMatchCount = (regexp: RegExp, searchString: string) : number => {
+        regexp.lastIndex = 0;
+        let retVal = 0;
+        while (regexp.test(searchString)) {
+            retVal++;
+        }
+        regexp.lastIndex = 0;
+        return retVal;
+    };
+
+    // First quickly make sure we could match up the URL to an existing item.
+    for (let urlToUpdate of newPrimaryUrlArray) {
+        const dependency = findDependencyToUpdate(urlToUpdate);
+        if (dependency.fallbackUrl) {
+            const dottedMatches : number = getMatchCount(dottedVersionRegExp, dependency.fallbackUrl);
+            const dashedMatches : number = getMatchCount(dashedVersionRegExp, dependency.fallbackUrl);
+            const matchCount : number = dottedMatches + dashedMatches;
+
+            if (matchCount == 0) {
+                throw new Error(`Version number not found in fallback URL '${dependency.fallbackUrl}'.`);
+            }
+            if (matchCount > 1) {
+                throw new Error(`Ambiguous version pattern found in fallback URL '${dependency.fallbackUrl}'. Multiple version strings found.`);
+            }
+        }
+    }
+
+    // Next take another pass to try and update to the URL  
+    const eventStream = new EventStream();
+    eventStream.subscribe((event: Event.BaseEvent) => {
+        switch (event.constructor.name) {
+            case Event.DownloadFailure.name:
+                console.log("Failed to download: " + (<Event.DownloadFailure>event).message);
```

Comment:
Like in the other places in the code, we could use a single eventStream that is passed to all these functions and in this file we could post the appropriate events to this stream and separately create a "ConsoleObserver" or "UpdatePackageDependencies" observer that subscribes to this event stream and does the logging accordingly.

Reply:
While I think the eventStream pattern that we created in other places make a lot of sense for enabling unit tests, I don't really see a benefit in this case since unit tests don't seem to make sense for a gulp task that produces output that would be code reviewed.